### PR TITLE
Allow IconButton to receive loading props

### PR DESCRIPTION
**Pull request by vascofg**
## Purpose

Allow IconButton to display a loading state.

## Approach and changes

Update types to allow `isLoading` and `loadingLabel` props for `IconButton`.
The behaviour already works from the parent `Button` component.

## Definition of done

* [X] Development completed
* [X] Reviewers assigned
* [ ] Unit and integration tests
* [ ] Meets minimum browser support
* [ ] Meets accessibility requirements

### CODE REVIEW

**Review by sumius — APPROVED** [review]
Good to me!

**Comment by robinmetral** [comment]
Hey @vascofg, thanks for the PR! Can you explain what you're trying to achieve a bit more? Since the prop types extend `ButtonProps`, as you mentioned, the loading state should already be supported. Isn't it?

**Comment by vascofg** [comment]
> Hey @vascofg, thanks for the PR! Can you explain what you're trying to achieve a bit more? Since the prop types extend `ButtonProps`, as you mentioned, the loading state should already be supported. Isn't it?

Hey @robinmetral, yes it is supported and it works, but generates a typescript error because the `isLoading` and the `loadingLabel` props are not allowed (marked as `never`).

So my PR does not change any implementation data, just updates the types.

File: .changeset/long-buttons-burn.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sumup/circuit-ui': patch
+---
+
+Allow IconButton to have a loading state
```
Comment (robinmetral) [inline .changeset/long-buttons-burn.md:5]: Allowed the loading state for the `IconButton` component. Pass an `isLoading` prop with an accessible `loadingLabel` label.

**Review by robinmetral — APPROVED** [review]
Ah my bad, somehow I misread this as _removing_ the loading state 🤦 makes sense to me, let's :shipit:

**Comment by robinmetral** [comment]
@vascofg just tweaked the changeset a bit, I'll let you press the big green button and then handle the new version release 🙂

**Pull request merged**